tests/online_migration/sle12_online_migration/snapper_rollback.pm

```diff
@@ -0,0 +1,79 @@
+# SLE12 online migration tests
+#
+# Copyright © 2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+use base "consoletest";
+use strict;
+use testapi;
+use utils;
+
+sub boot_snapshot_before_migration() {
+    assert_screen "grub2";
+    # prevent grub2 timeout
+    send_key "up";
+    send_key_until_needlematch("boot-menu-snapshot", 'down', 10, 5);
+    send_key "ret";
+    assert_screen("boot-menu-snapshot-list");
+    send_key_until_needlematch("select-snapshot-before-migration", 'down', 10, 5);
+    send_key "ret";
+    # bsc#956046  check if we are in first menu-entry, or not
+    # set timeout 5 to reduce check time
+    if (check_screen("boot-menu-snapshot-bootmenu", 5)) {
+        send_key 'down', 1;
+        save_screenshot;
+    }
+    send_key "ret";
+}
+
+sub snapper_rollback() {
+    assert_screen 'linux-login', 200;
+    # select_console works not well here, login as root directly
+    type_string "root\n";
+    assert_screen "password-prompt";
+    type_password;
+    send_key "ret";
+    wait_still_screen;
+    script_run "snapper rollback";
+}
+
+sub check_rollback_system() {
+    # first to check rollback-helper service is enabled and worked properly
+    # script_run "systemctl status rollback.service";
+    my $output = script_output "systemctl status rollback.service";
+    if ($output !~ /enabled.*?code=exited,\sstatus=0\/SUCCESS/s) {
+        die "rollback service was failed";
+    }
+
+    # second to check if repos were rolled back to original
+    script_run("zypper lr -u | tee /dev/$serialdev");
+}
+
+sub run() {
+    my $self = shift;
+    select_console 'root-console';
+
+    # reboot system after online migration
+    script_run("systemctl reboot", 0);
+
+    boot_snapshot_before_migration;
+    snapper_rollback;
+
+    # reboot into the system before online migration
+    script_run("systemctl reboot", 0);
+    wait_boot;
+    select_console 'root-console';
+
+    check_rollback_system;
+}
+
+sub test_flags {
+    return {important => 1};
+}
+
+1;
+# vim: set sw=4 et:
```

Comment:
@sysrich, As soon as I know all the hdd images for sp0 and sp1 are registered with the well known keys.
The keys are somewhere in the wiki.microfocus.net